tests/spirv/flat.slang

```diff
@@ -0,0 +1,27 @@
+//TEST:SIMPLE(filecheck=CHECK): -target spirv
+
+// Check that we don't emit duplicate Flat decorations for the same variable.
+
+// CHECK: OpDecorate %vo_a Flat
+// CHECK-NOT: OpDecorate %vo_a Flat
+
```

Comment:
This issue can be reproduced with an existing test writting in GLSL. `tests/glsl/flat-in.slang`
Maybe we can have the same check there too.

Reply:
That test is written in GLSL and intended for glsl compatibility only.
So let's keep the test here to dedicate for correctness purpose.